depthai_nodes/ml/parsers/parser.py

```diff
@@ -1,90 +1,57 @@
+    def build(
+        self, nn_archive: dai.NNArchive, head__index: int = None
+    ) -> List[BaseParser]:
+        """Instantiates parsers based on the provided model archive.
 
         Attributes
         ----------
-        heads : Union[List, Dict]
-            List of all archive head objects: [<depthai.nn_archive.v1.Head object>, ...]
-            or a dictionary containing the head metadata.
-        head_name : str
-            The name of the head to use. If multiple heads are available in the archive, the name must be specified.
+        nn_archive: dai.NNArchive
+            NN Archive of the model.
+        head__index: int
+            Index of the head to be used for parsing. If not provided, each head will instantiate a separate parser.
 
         Returns
         -------
-        Parser
-            Returns the parser object with the head configuration set.
+        parsers: List[BaseParser]
+            List of instantiated parsers.
         """
+        heads = nn_archive.getConfig().getConfigV1().model.heads
 
-        if isinstance(heads, list):
-            if len(heads) == 0:
-                raise ValueError("No heads available in the nn_archive.")
-
-            head = heads[0]
-
-            if len(heads) > 1 and head_name == "":
-                current_parser = self.__class__.__name__
-                parser_names_in_archive = [head.parser for head in heads]
-                num_matches = parser_names_in_archive.count(current_parser)
-                if num_matches == 0:
-                    raise ValueError(
-                        f"No heads available for {current_parser} in the nn_archive."
-                    )
-                elif num_matches == 1:
-                    head = [head for head in heads if head.parser == current_parser][0]
-                else:
-                    raise ValueError(
-                        f"Multiple heads with parser= {current_parser} detected, please specify a head name."
-                    )
+        if len(heads) == 0:
+            raise ValueError("No heads defined in the NN Archive.")
 
-            if head_name != "":
-                head_candidates = [
-                    head
-                    for head in heads
-                    if head.metadata.extraParams["name"] == head_name
-                ]
-                if len(head_candidates) == 0:
-                    raise ValueError(
-                        f"No head with name {head_name} specified in nn_archive."
-                    )
-                if len(head_candidates) > 1:
-                    raise ValueError(
-                        f"Multiple heads with name {head_name} found in nn_archive, please specify a unique name."
-                    )
-                head = head_candidates[0]
+        if head__index:
+            heads = [heads[head__index]]
+        parsers = []
+        pipeline = self.getParentPipeline()
 
+        for head in heads:
             parser_name = head.parser
```

Comment:
I added indexes before so in case the head_index is given, the key is that index insted of it being 0.